tools/clang/include/clang/AST/Decl.h

```diff
@@ -810,9 +810,14 @@ class VarDecl : public DeclaratorDecl, public Redeclarable<VarDecl> {
     /// declared.
     unsigned ScopeDepthOrObjCQuals : 7;
 
+    // HLSL Change start
     /// Whether the parameter is copied out.
     unsigned IsModifierOut : 1;
 
+    // Index into allowable parameter conversions
+    unsigned LegalTypesIdx : 6;
```

Comment:
It should be 8 even though no value goes higher than 40 at the moment. It's 6 because that's all the bits we have left of 32. If this is validated as the right way to do this, I'll work out a more robust representation.